src/init/projectInit.ts

```diff
@@ -141,57 +73,18 @@ export class ProjectInitializer extends Disposable {
             this.handleError(p, "Failed to create template. " + err.message, projectFolder.fsPath);
         });
-        proc.addListener('close', async () => await p.end(projectFolder));
-    }
-
-    private getTemplatesList(): Array<FileListItem> {
-        return this._templates.map((tmpl) => new FileListItem(tmpl.name, tmpl.desc, tmpl.name + ".zip"));
-    }
-
-    private async createFromTemplate(tmpl: FileListItem, destUri: Uri, p: ProgressHandler) {
-        let tmpDir = this.createTempDir();
-        let tmpFilePath = path.join(tmpDir, tmpl.value);
-        p.report('Downloading template...');
-        await this.downloadTemplateAndSetup(tmpl, tmpFilePath, tmpDir, destUri, p);
-    }
-
-    private async downloadTemplateAndSetup(tmpl: FileListItem, tmpFilePath, tmpDir, destUri: Uri, p: ProgressHandler) {
-        let req = get(`${GAUGE_TEMPLATE_URL}/${tmpl.value}`, (res) => {
-            res.on('data', (d) => fs.appendFileSync(tmpFilePath, d));
-            res.on('end', async () => await this.extractZipAndCopyFiles(tmpFilePath, tmpDir, tmpl.label, destUri, p));
-        });
-        req.on('error', (err) => {
-            this.handleError(p, "Failed to download template. " + err.message, destUri.fsPath);
+        proc.stdout.on('data', (m) => { console.log(m.toString()); });
+        proc.on('close', async (code) => {
+            if (code === 0) { p.cancel("Faile to initialize project."); }
+            await p.end(projectFolder);
         });
     }
 
-    private async extractZipAndCopyFiles(zip, tmpDir, tmpl: string, destUri: Uri, p: ProgressHandler) {
-        try {
-            p.report("Extracting template...");
-            new AdmZip(zip).extractAllTo(tmpDir, true);
-            p.report("Copying files to " + destUri.fsPath);
-            await fs.copy(path.join(tmpDir, tmpl), destUri.fsPath);
-            this.runPostInstall(destUri.fsPath, p);
-            p.end(destUri);
-        } catch (err) {
-            return this.handleError(p, "Failed to extract template. " + err.message, destUri.fsPath);
-        }
-    }
-
-    private runPostInstall(projectFolder: string, p: ProgressHandler) {
-        let metadataFile = path.join(projectFolder, 'metadata.json');
-        let cmd = JSON.parse(fs.readFileSync(metadataFile, 'UTF-8')).postInstallCmd;
-        fs.unlinkSync(metadataFile);
-        if (!cmd) return;
-        p.report('Running post install commands to setup project...');
-        execSync(cmd, { cwd: projectFolder });
-    }
-
-    private createTempDir() {
-        let tmpDir = path.join(os.tmpdir(), 'gauge');
-        fs.removeSync(tmpDir);
-        fs.mkdirSync(tmpDir);
-        return tmpDir;
+    private async getTemplatesList(): Promise<Array<FileListItem>> {
+        let args = ["template", "--list", "--machine-readable"];
+        let cp = spawnSync(this.cli.gaugeCommand(), args, { env: process.env });
+        let _templates = JSON.parse(cp.stdout.toString());
```

Comment:
JSON.parse can throw an error, and this being an async method, will result in a UnhandledPromiseRejection error.